client/shared/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -38,6 +39,14 @@ import scala.math.random
 object Retry {
   private[this] val logger = getLogger
 
+  /** This key tracks the attempt count, the retry middleware starts the very
+    * first request with 1 as the first request, even with no retries
+    * (aka the first attempt). If one wants to monitor retries
+    * explicitly and not the attempts one may want to subtract the
+    * value of this key by 1.
+    */
+  val AttemptCountKey: Key[Int] = Key.newKey[cats.effect.SyncIO, Int].unsafeRunSync()
```

Comment:
Just wondering, can we mark this lazy?

Reply:
Hmm, do we have a precedent for making keys lazy in http4s? I don't think it's very expensive, and it's already lazy depending on whether the relevant classes are loaded or not anyway.

Reply:
Yeah, that's correct. No key is defined as 'lazy' at the moment (ok, most of those I've seen). But is it intended?

Reply:
Though, my comment isn't about this particular PR, and don't hesitate on merging.

Reply:
> But is it intended?

IMHO I think it's fine. Allocating a key is cheap, and it will only happen if the middleware class is loaded into the JVM anyway because of lazy class initialization. Meanwhile, any `lazy val` adds an additional synchronization point which probably has some performance cost over vanilla `val`.